addons/mail/static/src/components/attachment/attachment.js

```diff
@@ -119,6 +119,11 @@ class Attachment extends Component {
      */
     _onClickUnlink(ev) {
         ev.stopPropagation();
+        if (this.attachment.isTemporary) {
+            this.env.messagingBus.trigger('o-attachment-upload-abort', {
+                attachment: this.attachment
+            });
+        }
```

Comment:
Why not in the `.catch()` of file uploader, in case of actual abort?

Reply:
Then we should find a way in `mockFetch` to throw an `AbortError` when `abortController.abort()` is called as the mock does not handle that feature I think

Reply:
Indeed. I think it's worth making this change in mocked fetch, so that the abort even is triggered exactly where it's appropriate.

Reply:
I'm not really sure it's doable by the way. In mock we only control the endpoint so we can only ask it to never return.
Here we need fetch to abort a non-existing request which results in the error to not be fired as there is nothing to abort.

So what you ask is +/- to mock browser.fetch ... is it really worthy ?

Reply:
mocked `browser.fetch` just calls `performFetch` of mock server, so if the manage of abort controller is managed in mock server performFetch method, so is mocked `browser.fetch`.

Definitely worth it IMO: it's the test that should be as close to reality as possible, not the actual code that must do changes to make test happy, especially when it weakens the code. Triggering of event that an uploading has been aborted should be exactly where the uploading is actually aborted, not somewhere else in the code. Because of this, any change in flow could break the correctness of triggering this event. It's equivalent to using heuristics: it makes code super weak.

Reply:
I really don't know how you could achieve what you want as I can't manage the AbortError to be fired anyhow.
I found that AbortSignal allow to define an "onabort" handler and using it makes the test pass without any change in mock fetch so I think it's good.

If you still think we should do something in mockfetch then feel free to write something because I don't know how we can achieve what you have in mind

Reply:
I'm good with the use of `onabort`, in that this code is closer to where it should be (`models/mail.attachment` instead `components/FileUploader`). Thanks ;)